tdrs-backend/tdpservice/parsers/case_consistency_validator.py

```diff
@@ -0,0 +1,136 @@
+                                 "Skipping Cat4 validation.")
+            return num_errors
+        except Exception as e:
+            logger.error(f"Uncaught exception during category four validation: {e}")
+            return num_errors
+
+    def __validate(self):
+        """Private validate, lint complexity."""
+        num_errors = 0
+        self.total_cases_validated += 1
+        self.has_validated = True
+        logger.debug(f"Attempting to execute Cat4 validation for case: {self.current_case}.")
+        if self.program_type == "TAN" and self.section == "A" and "state_fips" in self.header:
+            num_errors += self.__validate_tanf_s1_case(num_errors)
+        elif self.program_type == "TAN" and self.section == "C" and "state_fips" in self.header:
+            num_errors += self.__validate_tanf_s2_case(num_errors)
+        elif self.program_type == "TAN" and self.section == "A" and "tribe_code" in self.header:
+            num_errors += self.__validate_tribal_tanf_s1_case(num_errors)
+        elif self.program_type == "TAN" and self.section == "C" and "tribe_code" in self.header:
+            num_errors += self.__validate_tribal_tanf_s2_case(num_errors)
+        elif self.program_type == "SSP" and self.section == "A":
+            num_errors += self.__validate_ssp_s1_case(num_errors)
+        elif self.program_type == "SSP" and self.section == "C":
+            num_errors += self.__validate_ssp_s2_case(num_errors)
+        else:
+            self.total_cases_validated -= 1
+            logger.warn(f"Case: {self.current_case} has no errors but has either an incorrect "
+                        f"program type: {self.program_type} or an incorrect section: {self.section}. No "
+                        "validation occurred.")
+            self.has_validated = False
+        return num_errors
+
+    def __validate_tanf_s1_case(self, num_errors):
+        """Perform TANF Section 1 category four validation on all cached records."""
+        num_errors += self.__validate_tanf_s1_header_with_records()
+        return num_errors
+
+    def __validate_tanf_s2_case(self, num_errors):
+        """Perform TANF Section 2 category four validation on all cached records."""
+        return 0
+
+    def __validate_tribal_tanf_s1_case(self, num_errors):
+        """Perform Tribal TANF Section 1 category four validation on all cached records."""
+        return 0
+
+    def __validate_tribal_tanf_s2_case(self, num_errors):
+        """Perform Tribal TANF Section 2 category four validation on all cached records."""
+        return 0
+
+    def __validate_ssp_s1_case(self, num_errors):
+        """Perform SSP Section 1 category four validation on all cached records."""
+        return 0
+
+    def __validate_ssp_s2_case(self, num_errors):
+        """Perform SSP Section 2 category four validation on all cached records."""
+        return 0
+
+    def __validate_tanf_s1_header_with_records(self):
```

Comment:
@elipe17 what about this one? 🆙

Reply:
@raftmsohani

Reply:
Yes, you are correct, but 2799 throws preparsing error while this is a parsing error, so there is no conflict. I am more leaning to keep this

Reply:
If a case has any errors associated with it, cat1/2/3, then cat4 validation is skipped. So if the the error in 2799 is generated, there will not be a duplicate generated from the cat4 section. With that said, we could remove this validator since it should be impossible for the cat4 one to be generated. If that doesn't cause too many headaches for @jtimpe in his downstream PRs, I would be ok with removing this validator and its associated tests.